### PR TITLE
Update config.mts to fix links & add Delete post, typing, and ulist

### PR DESCRIPTION
**Pull request by Finley224**
The docs have broken links that go to the old mower media org and the wrong repos, this commit should fix that.

### CODE REVIEW

**Comment by Finley224** [comment]
I know there is another pull request, but this one only fixes the links and does nothing else.

**Comment by Finley224** [comment]
New addition:
I also moved some stuff around on the sidebar, instead of having a list of all the cloudlink stuff, I think it is better to group it into packets (things the server sends), and commands (things the client sends to the server).

File: docs/api/cloudlink/intro.md [hunk 1]
```diff
@@ -8,4 +8,3 @@
     - The general request/response shape
 -->
 
-This section is out of date.
```
Comment (mybearworld) [inline docs/api/cloudlink/intro.md:11]: This section is still out of date

File: docs/api/cloudlink/packets/ulist.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+This packet provides a string of online users separated by ";"
+## Payload
+
+A string of users separated by ";"
+```
+user1;user2;user3;
+```
+
+## Usage
+::: code-group
```
Comment (mybearworld) [inline docs/api/cloudlink/packets/ulist.md]: 1. Is this section necessary at all? 
2. All the code blocks should remove the last element as well as the ulist puts a semicolon at the end

File: docs/api/cloudlink/packets/ulist.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+```ts [TypeScript]
+// Converts the online userlist into a array of strings
+// val is the string of users given by the websocket
+let onlineUsers: string[] = val.split(";");
```
Comment (mybearworld) [inline docs/api/cloudlink/packets/ulist.md]: The JS code works in Typescript as well and is more idiomatic

File: docs/api/cloudlink/intro.md (hunk 1 above)
Comment (mybearworld) [reply docs/api/cloudlink/intro.md:11]: ...which makes the rest of the added Cloudlink sections a little confusing as those are *not* out of date

File: .vitepress/sidebar.mts [hunk 4]
```diff
@@ -9,20 +9,44 @@ export const sidebar: DefaultTheme.SidebarItem[] = [{
...
 			collapsed: true,
 			items: [
 				{
 					text: 'Introduction',
 					link: '/api/cloudlink/intro',
 				},
 				{
-					text: 'authpswd command',
-					link: '/api/cloudlink/commands/authpswd',
+					text: 'Commands',
```
Comment (mybearworld) [inline .vitepress/sidebar.mts]: The only command that still exists with ?v=1 is `ping` iirc - does there need to be a separate section for that?

**Comment by mybearworld** [comment]
https://github.com/meower-media/docs/blob/d87728fcad9652ac9ca5984b51c0b40b7792d805/docs/api/cloudlink/packets/typing.md?plain=1#L4-L7

Interesting, I'm not sure where I got the 3s time from... Meower Svelte does use 4s and the server has them limited to six in five seconds.

File: .vitepress/sidebar.mts (hunk 4 above)
Comment (Finley224) [reply .vitepress/sidebar.mts]: good point, let me change that.

File: docs/api/cloudlink/packets/ulist.md (hunk 3 above)
Comment (Finley224) [reply docs/api/cloudlink/packets/ulist.md]: Ok, I will merge them into one.

File: docs/api/cloudlink/packets/ulist.md (hunk 2 above)
Comment (Finley224) [reply docs/api/cloudlink/packets/ulist.md]: 1. Probably not, I can remove it.

File: docs/api/cloudlink/intro.md (hunk 1 above)
Comment (Finley224) [reply docs/api/cloudlink/intro.md:11]: What about " this section is incomplete"?

Comment (mybearworld) [reply docs/api/cloudlink/intro.md:11]: That would be misleading, information that is on here would contradict itself right now

Comment (Finley224) [reply docs/api/cloudlink/intro.md:11]: I added a list of the up to date sections, does that work?

Comment (Finley224) [reply docs/api/cloudlink/intro.md:11]: I added a list of the up to date sections, does that work?

File: .vitepress/sidebar.mts [hunk 5]
```diff
@@ -9,20 +9,40 @@ export const sidebar: DefaultTheme.SidebarItem[] = [{
 			link: 'https://docs.meower.org/api/rest.html',
 		},
 		{
-			text: 'Cloudlink',
+			text: 'Cloudlink (websocket)',
```
Comment (williamhorning) [inline .vitepress/sidebar.mts:12]: there's no reason to change this, everything still refers to it as cloudlink. we will eventually change the name when we start renaming all the services

File: docs/api/cloudlink/intro.md [hunk 6]
```diff
@@ -8,4 +8,9 @@
     - The general request/response shape
 -->
```
Comment (williamhorning) [inline docs/api/cloudlink/intro.md:10]: the points in the comment should be addressed either here or in a new pr, the current changes here aren't that great imo

File: docs/api/cloudlink/intro.md [hunk 7]
```diff
@@ -8,4 +8,9 @@
...
 
-This section is out of date.
+- This page is out of date.
+
+Up to date pages:
```
Comment (williamhorning) [inline docs/api/cloudlink/intro.md:13]: i don't really like this approach and again, something should update these

File: docs/api/cloudlink/packets/delete_post.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# `delete_post`
+
+This packet is sent when a is deleted, when a client gets this packet, it should search any local cache for the post_id and remove any matching posts.
```
Comment (williamhorning) [inline docs/api/cloudlink/packets/delete_post.md]:
```suggestion
This packet is sent when a post is deleted. 

```

generally, these docs are intended to be a reference and shouldn't tell clients what to do, that's what the intent of the apic examples and stuff are for

File: docs/api/cloudlink/packets/typing.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# `typing`
+
+This packet is sent when a client sends a typing indicator. These clients will be sent periodically by clients, so you need to setup a expiration system.
+### Client typing indicator frequency
+- **meo**: 3 seconds
+- **Roarer 2**: 3 seconds
+- **Meower Svelte**: 4 seconds
+
```
Comment (williamhorning) [inline docs/api/cloudlink/packets/typing.md:8]:
```suggestion
This packet is sent when a client is currently typing.

```

again, client behavior shouldn't go here

File: docs/api/cloudlink/packets/ulist.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# `ulist`
+
+This packet provides a string of online users separated by ";"
```
Comment (williamhorning) [inline docs/api/cloudlink/packets/ulist.md:3]:
```suggestion
This packet contains a comma-separated list of all online users

```

**Review by williamhorning — CHANGES_REQUESTED** [review]
in addition to the feedback i have below(?), please run deno fmt and make sure the checks pass

File: docs/api/cloudlink/packets/typing.md (hunk 9 above)
Comment (mybearworld) [reply docs/api/cloudlink/packets/typing.md:8]: There should probably still be some sort of reference on what "periodically" means though.

File: .vitepress/sidebar.mts (hunk 5 above)
Comment (Finley224) [reply .vitepress/sidebar.mts:12]: Ok, when I get to my computer I will change that.

File: docs/api/cloudlink/packets/ulist.md (hunk 10 above)
Comment (Finley224) [reply docs/api/cloudlink/packets/ulist.md:3]: Ok, I will update that.

File: docs/api/cloudlink/intro.md (hunk 6 above)
Comment (Finley224) [reply docs/api/cloudlink/intro.md:10]: I was planning to do that when I get time, do you want me to wait to do the pull request until it's done?

File: docs/api/cloudlink/intro.md (hunk 7 above)
Comment (Finley224) [reply docs/api/cloudlink/intro.md:13]: Makes sense, I will update that.

File: docs/api/cloudlink/packets/typing.md (hunk 9 above)
Comment (Finley224) [reply docs/api/cloudlink/packets/typing.md:8]: I disagree on this one, it's a helpful reference to have the typing times in the API docs as there is no standard for it. What if I remove the note about how to implement it but leave a general note of most clients send the packet at least every 4 seconds?

**Comment by Finley224** [comment]
> in addition to the feedback i have below(?), please run deno fmt and make sure the checks pass

Ok, when I'm back at my computer I will do that.

File: docs/api/cloudlink/packets/typing.md (hunk 9 above)
Comment (williamhorning) [reply docs/api/cloudlink/packets/typing.md:8]: the lack of standard is the problem here, but i'm not opposed to saying "send a typing indicator every few seconds" note

Comment (showierdata9978) [reply docs/api/cloudlink/packets/typing.md:8]: Maybe say it expires after 4 seconds

Comment (mybearworld) [reply docs/api/cloudlink/packets/typing.md:8]: > Maybe say it expires after 4 seconds

Only Meower Svelte seems to use that though

Comment (Finley224) [reply docs/api/cloudlink/packets/typing.md:8]: > Maybe say it expires after 4 seconds



> > Maybe say it expires after 4 seconds
> 
> Only Meower Svelte seems to use that though

I like the 4s idea, all the other ones seem to be less than 4s so 4s would work.

**Comment by Finley224** [comment]
I will fix the errors with the checks soon.

File: docs/api/cloudlink/intro.md (hunk 7 above)
Comment (Finley224) [reply docs/api/cloudlink/intro.md:13]: I will update this when I get to it, just don't have time at the moment, do you have an idea on how the message should be written @williamhorning ?

File: docs/api/cloudlink/intro.md (hunk 6 above)
Comment (Finley224) [reply docs/api/cloudlink/intro.md:10]: I will do that in a new pr when I get time.

File: docs/api/cloudlink/packets/typing.md (hunk 9 above)
Comment (Finley224) [reply docs/api/cloudlink/packets/typing.md:8]: does the 4s thing work for you @williamhorning?

Comment (williamhorning) [reply docs/api/cloudlink/packets/typing.md:8]: @Finley224 either saying "every four seconds" or saying "at an interval less than five seconds" is fine

File: docs/api/cloudlink/intro.md (hunk 7 above)
Comment (williamhorning) [reply docs/api/cloudlink/intro.md:13]: Actually, it seems as though most the only thing not up to date would be the authpswd page, which we could just add a notice box to. we should probably add a brief introduction to cloudlink on this page though

**Comment by williamhorning** [comment]
merged this in d76ad95fbbb6591735ea350997e81d8e01332657

**Pull request closed without merging**